src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -121,33 +121,42 @@ public void afterSave() {
-                                "Server config not found for input server id " + getServerId()));
-                List<BitbucketWebhookMultibranchTrigger> triggers = getTriggers(project);
-                boolean isPullRequestTrigger =
-                        triggers.stream().anyMatch(BitbucketWebhookMultibranchTrigger::isPullRequestTrigger);
-                boolean isRefTrigger = triggers.stream().anyMatch(BitbucketWebhookMultibranchTrigger::isRefTrigger);
-
-                try {
-                    descriptor.getRetryingWebhookHandler().register(
-                            bitbucketServerConfiguration.getBaseUrl(),
-                            bitbucketServerConfiguration.getGlobalCredentialsProvider(owner),
-                            repository, owner, isPullRequestTrigger, isRefTrigger);
-                } catch (WebhookRegistrationFailed webhookRegistrationFailed) {
-                    LOGGER.severe("Webhook failed to register- token credentials assigned to " +
-                            bitbucketServerConfiguration.getServerName()
-                            +
-                            " do not have admin access. Please reconfigure your instance in the Manage Jenkins -> Settings page.");
-                }
+        SCMSourceOwner owner = getOwner();
+        if (!(owner instanceof ComputedFolder)) {
+            return;
+        }
+
+        ComputedFolder<?> project = (ComputedFolder<?>) owner;
+        if (!webhookRegistered && isValid()) {  
+            DescriptorImpl descriptor = (DescriptorImpl) getDescriptor();
+            BitbucketServerConfiguration bitbucketServerConfiguration = descriptor.getConfiguration(getServerId())
+                    .orElseThrow(() -> new BitbucketClientException(
+                            "Server config not found for input server id " + getServerId()));
+            List<BitbucketWebhookMultibranchTrigger> triggers = getTriggers(project);
+            boolean isPullRequestTrigger =
+                    triggers.stream().anyMatch(BitbucketWebhookMultibranchTrigger::isPullRequestTrigger);
+            boolean isRefTrigger = triggers.stream().anyMatch(BitbucketWebhookMultibranchTrigger::isRefTrigger);
+
+            try {
+                descriptor.getRetryingWebhookHandler().register(
+                        bitbucketServerConfiguration.getBaseUrl(),
+                        bitbucketServerConfiguration.getGlobalCredentialsProvider(owner),
+                        repository, owner, isPullRequestTrigger, isRefTrigger);
+            } catch (WebhookRegistrationFailed webhookRegistrationFailed) {
+                LOGGER.severe("Webhook failed to register- token credentials assigned to " +
+                        bitbucketServerConfiguration.getServerName()
+                        +
+                        " do not have admin access. Please reconfigure your instance in the Manage Jenkins -> Settings page.");
             }
         }
+
+        try{
+            project.save();
```

Comment:
This saves the configuration to disk. This could have side effects... However, it should prevent us from having to re-initialize the webhook each time since the state the webhook looks for is already present.